src/service.rs

```diff
@@ -430,6 +438,28 @@ mod test {
          event_rx)
     }
 
+    fn timebomb<R, F>(dur: Duration, f: F) -> R
```

Comment:
I'd find this a lot easier to read with an explaining comment, or maybe a more informative name. (`run_with_timeout`?)